subprojects/platform-play/src/main/java/org/gradle/play/internal/twirl/TwirlCompilerAdapterV13X.java

```diff
@@ -0,0 +1,150 @@
+/*
+ * Copyright 2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.play.internal.twirl;
+
+import org.gradle.language.twirl.TwirlImports;
+import org.gradle.language.twirl.TwirlTemplateFormat;
+import org.gradle.scala.internal.reflect.ScalaCodecMapper;
+import org.gradle.scala.internal.reflect.ScalaMethod;
+import org.gradle.scala.internal.reflect.ScalaReflectionUtil;
+import org.gradle.util.CollectionUtils;
+
+import java.io.File;
+import java.lang.reflect.InvocationTargetException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+
+class TwirlCompilerAdapterV13X extends TwirlCompilerAdapterV10X {
+
+    private static final Iterable<String> SHARED_PACKAGES = Arrays.asList("play.twirl.compiler", "scala.io", "scala.util.parsing.input", "scala.collection");
+
+    // Default imports are based on:
+    // https://github.com/playframework/playframework/blob/2.6.0/framework/src/build-link/src/main/java/play/TemplateImports.java
+    private static List<String> DEFAULT_JAVA_TEMPLATE_IMPORTS;
+    private static List<String> DEFAULT_SCALA_TEMPLATE_IMPORTS;
+
+    private static List<String> DEFAULT_TEMPLATE_IMPORTS = Collections.unmodifiableList(
+        Arrays.asList(
+            "models._",
+            "controllers._",
+            "play.api.i18n._",
+            "play.api.templates.PlayMagic._",
+            // Based on https://github.com/playframework/twirl/blob/1.3.13/compiler/src/main/scala/play/twirl/compiler/TwirlCompiler.scala#L156
```

Comment:
We need these additional comments to ensure users can write twirl templates like:

```html
@defining(10) { v =>
   The defined value is @v
}
```
This was not working for Play 2.6/Twirl 1.3 because the `defining` function was not imported.